internal/manifests/targetallocator/configmap.go

```diff
@@ -93,3 +104,78 @@ func ConfigMap(params Params) (*corev1.ConfigMap, error) {
 		},
 	}, nil
 }
+
+func getGlobalConfig(taGlobalConfig v1beta1.AnyConfig, collectorConfig v1beta1.Config) (map[string]any, error) {
+	// global config from the target allocator has priority
+	if len(taGlobalConfig.Object) > 0 {
+		return taGlobalConfig.Object, nil
+	}
+
+	collectorGlobalConfig, err := getGlobalConfigFromOtelConfig(collectorConfig)
+	if err != nil {
+		return nil, err
+	}
+	return collectorGlobalConfig.Object, nil
+}
+
+func getScrapeConfigs(taScrapeConfigs []v1beta1.AnyConfig, collectorConfig v1beta1.Config) ([]v1beta1.AnyConfig, error) {
+	scrapeConfigs := []v1beta1.AnyConfig{}
+
+	// we take scrape configs from both the target allocator spec and the collector config
+	if len(taScrapeConfigs) > 0 {
+		scrapeConfigs = append(scrapeConfigs, taScrapeConfigs...)
+	}
```

Comment:
of note – if someone just creates a targetallocator crd they no longer need to escape scrape configs... this may make it easier for us to use upstream libraries for parsing prometheus in the future.